CodeEdit/Features/Git/Client/GitClient+Remote.swift

```diff
@@ -31,6 +31,20 @@ extension GitClient {
     func removeRemote(name: String) async throws {
         _ = try await run("remote rm \(name)")
     }
+
+    /// Get the URL of the remote
+    /// > Note: If a git repository has multiple remotes, by default the `origin` remote
+    /// > will be used, unless there’s an upstream branch configured for the current branch.
+    /// > (Reference: https://git-scm.com/docs/git-ls-remote, https://git-scm.com/docs/git-fetch)
+    /// - Returns: A URL if a remote is configured, nil otherwise
+    func getRemoteURL() async -> URL? {
```

Comment:
I'm definitely open to comments on this function. A few things I'm not 100% on:

1. Is my approach idiomatic Swift? Would a seasoned Swift programmer do it differently? (I am not a seasoned Swift programmer!) For example, the explicit `do / catch` construct feels a bit old skool; maybe you'd prefer `guard let remote = try? await ...` instead?
2. The other functions in this extension all `throw`. I chose not to do that here, because in my experience throwing in a function that returns an optional leads to less readable code; there are two, different non-happy-path states for callers to reason about. But I believe equally (more?) strongly that it's important for codebases to be consistent. So I'm eager to hear your thoughts on this.

Reply:
You raise very good points, I'll leave my thoughts for each:

1. I personally like the explicit `do/catch` wrap. It makes it very legible that this method explicitly catches its errors and does something with them. The `try?` keyword on the other hand is explicitly ignoring some information about why the callee is failing, resulting in lost information. 
2. For this method, I think a throwing and return optional method would be more appropriate (with a caveat). You're right that often that is less legible, but because this method will be called in an already-throwing method the error is always being passed to whoever is performing a larger operation. Therefore, by ignoring an error thrown here could result in a real error being ignored (like `git` moving or something). *So*, if we don't throw, we potentially lose information but we also need to indicate the lack of a value using an optional. So in this case (in my opinion) it should throw and return an optional. A `- throws:` doc comment could be useful to help discern what the two unhappy cases are.

The caveat is if there's no way to discern whether the error thrown is due to no remote existing, or because of some other error with the method, the error should be ignored. In that case it should just have a comment on the `do/catch` mentioning why the explicit error ignore.

Reply:
Thanks @thecoolwinter. Really good point about not wanting to ignore the error. I'll update this.